test/support/factories/trip_planner/trip_planner.ex

```diff
@@ -2,182 +2,201 @@ defmodule Test.Support.Factories.TripPlanner.TripPlanner do
   @moduledoc """
   Provides generated test data via ExMachina and Faker.
   """
+
   use ExMachina
 
   alias Dotcom.TripPlan.{NamedPosition, Parser}
   alias OpenTripPlannerClient.Test.Support.Factory
+  alias Test.Support.Factories.Stops.Stop
 
-  def itinerary_factory do
-    Factory.build(:itinerary)
-    |> limit_route_types()
-    |> Parser.parse()
-  end
-
-  def otp_itinerary_factory do
-    Factory.build(:itinerary)
-    |> limit_route_types()
-  end
-
-  def leg_factory do
-    [:walking_leg, :transit_leg]
-    |> Faker.Util.pick()
-    |> Factory.build()
-  end
-
-  def walking_leg_factory do
-    Factory.build(:walking_leg)
-    |> Parser.parse()
-  end
-
-  def transit_leg_factory do
-    Factory.build(:transit_leg)
-    |> limit_route_types()
+  def bus_leg_factory do
+    build(:otp_bus_leg)
     |> Parser.parse()
   end
 
-  def subway_leg_factory do
+  def cr_leg_factory do
     Factory.build(:transit_leg, %{
       agency: Factory.build(:agency, %{name: "MBTA"}),
       route:
         Factory.build(:route, %{
-          type: 1
+          type: 2
         })
     })
     |> Parser.parse()
   end
 
-  def bus_leg_factory do
+  def express_bus_leg_factory do
     Factory.build(:transit_leg, %{
       agency: Factory.build(:agency, %{name: "MBTA"}),
       route:
         Factory.build(:route, %{
+          gtfs_id: "mbta-ma-us:" <> Faker.Util.pick(Fares.express()),
           type: 3
         })
     })
     |> Parser.parse()
   end
 
-  def express_bus_leg_factory do
+  def ferry_leg_factory do
+    build(:otp_ferry_leg)
+    |> Parser.parse()
+  end
+
+  def groupable_otp_itinerary_factory do
+    [a, b, c] =
+      Enum.map(1..3, fn _ ->
+        Factory.build(:place, stop: Factory.build(:stop))
+      end)
```

Comment:
Would `Factory.build_list(3, :place, stop: Factory.build(:stop))` work just as well?

Reply:
Can happen later, I feel like this should be part of OpenTripPlannerClient's factory

Reply:
I don't think so because 'grouping' is a dotcom concept not an OTP client concept.